frontend/src/metabase/static-viz/components/WaterfallChart/WaterfallChart.jsx

```diff
@@ -0,0 +1,342 @@
+  accessors = getDefaultAccessors(type),
+  settings,
+  labels,
+  getColor,
+}) {
+  const chartStyle = {
+    fontFamily: "Lato, sans-serif",
+    axes: {
+      color: getColor("text-light"),
+      ticks: {
+        color: getColor("text-medium"),
+        fontSize: 12,
+      },
+      labels: {
+        color: getColor("text-medium"),
+        fontSize: 14,
+        fontWeight: 700,
+      },
+    },
+    value: {
+      color: getColor("text-dark"),
+      fontSize: 12,
+      fontWeight: 800,
+      stroke: getColor("white"),
+      strokeWidth: 3,
+    },
+  };
+
+  const axesProps = {
+    stroke: chartStyle.axes.color,
+    tickStroke: chartStyle.axes.color,
+  };
+
+  const valueProps = {
+    fontSize: chartStyle.value?.fontSize,
+    fontFamily: chartStyle.fontFamily,
+    fontWeight: chartStyle.value?.fontWeight,
+    letterSpacing: 0.5,
+    fill: chartStyle.value?.color,
+    stroke: chartStyle.value?.stroke,
+    strokeWidth: chartStyle.value?.strokeWidth,
+  };
+
+  if (type === "timeseries") {
+    data = sortTimeSeries(data);
+  }
+  const entries = calculateWaterfallEntries(
+    data,
+    accessors,
+    settings?.showTotal,
+  );
+
+  const isVertical = type === "timeseries" ? false : entries.length > 10;
+  const xTickWidth =
+    type === "timeseries"
+      ? // We don't know the width of the time-series label because it's formatted inside `<AxisBottom />`.
+        // We could extract those logic out, but it's gonna be nasty, and we didn't need `xTickWidth` for time-series anyway.
+        0
+      : getXTickWidth(
+          data,
+          accessors,
+          layout.maxTickWidth,
+          chartStyle.axes.ticks.fontSize,
+        );
+
+  const getXTickProps =
+    type === "timeseries"
+      ? ({ formattedValue, ...props }) => {
+          return {
+            ...props,
+            children: formatTimescaleWaterfallTick(formattedValue, settings),
+          };
+        }
+      : ({ x, y, formattedValue, ...props }) => {
+          const textWidth = isVertical ? xTickWidth : xScale.bandwidth();
+          const truncatedText = truncateText(
+            formattedValue,
+            textWidth,
+            chartStyle.axes.ticks.fontSize,
+          );
+          const xTickFontSize = chartStyle.axes.ticks.fontSize;
+          const transform = isVertical
+            ? `rotate(-90, ${x} ${y}) translate(${Math.floor(
+                xTickFontSize * 0.9,
+              )} ${Math.floor(xTickFontSize / 3)})`
+            : undefined;
+
+          const textAnchor = isVertical ? "end" : "middle";
+
+          return {
+            ...props,
+            x,
+            y,
+            transform,
+            children: truncatedText,
+            textAnchor,
+          };
+        };
+
+  const numTicks = type === "timeseries" ? layout.numTicks : entries.length;
+  const tickLabelProps = getXTickLabelProps(chartStyle, isVertical);
+  const topMargin = settings.show_values
+    ? layout.margin.top + VALUES_MARGIN
```

Comment:
Needs an extra top margin so we can display the data point value at the top of the chart without having it overflown.